examples/example-tls/src/main/java/io/grpc/examples/helloworldtls/HelloWorldServerS2A.java

```diff
@@ -0,0 +1,141 @@
+ * [privateKeyFilePath] File Path to Server Private Key
+ * [trustCertCollectionFilePath] File Path to CA Certificate
+ */
+public class HelloWorldServerS2A {
+    private static final Logger logger = Logger.getLogger(HelloWorldServerS2A.class.getName());
+    private Server server;
+    private final int port;
+    private final String certChainFilePath;
+    private final String privateKeyFilePath;
+    private final String trustCertCollectionFilePath;
+
+    private HelloWorldServerS2A(int port,
+                             String certChainFilePath,
+                             String privateKeyFilePath,
+                             String trustCertCollectionFilePath) {
+        this.port = port;
+        this.certChainFilePath = certChainFilePath;
+        this.privateKeyFilePath = privateKeyFilePath;
+        this.trustCertCollectionFilePath = trustCertCollectionFilePath;
+    }
+
+    private SslContextBuilder getSslContextBuilder() {
+      SslContextBuilder sslClientContextBuilder = SslContextBuilder.forServer(new File(certChainFilePath), new File(privateKeyFilePath));
+        if (trustCertCollectionFilePath != null) {
+            sslClientContextBuilder.trustManager(new File(trustCertCollectionFilePath));
+            sslClientContextBuilder.clientAuth(ClientAuth.REQUIRE);
+            sslClientContextBuilder.protocols("TLSv1.3");
+        }
+        return GrpcSslContexts.configure(sslClientContextBuilder);
+    }
+
+    private void start() throws IOException {
+        server = NettyServerBuilder.forPort(port)
+                .sslContext(getSslContextBuilder().build())
+                .addService(new GreeterImpl())
+                .build()
+                .start();
+
+        logger.info("Server started, listening on " + port);
+        Runtime.getRuntime().addShutdownHook(new Thread() {
+            @Override
+            public void run() {
+                // Use stderr here since the logger may have been reset by its JVM shutdown hook.
+                System.err.println("*** shutting down gRPC server since JVM is shutting down");
+                HelloWorldServerS2A.this.stop();
+                System.err.println("*** server shut down");
+            }
+        });
+    }
+
+    private void stop() {
+        if (server != null) {
+            server.shutdown();
+        }
+    }
+
+    /**
+     * Await termination on the main thread since the grpc library uses daemon threads.
+     */
+    private void blockUntilShutdown() throws InterruptedException {
+        if (server != null) {
+            server.awaitTermination();
+        }
+    }
+
+    private static HelloWorldServerS2A create(int port,
+                             String certChainFilePath,
+                             String privateKeyFilePath,
+                             String trustCertCollectionFilePath) {
+        checkNotNull(port, "port");
+        checkNotNull(certChainFilePath, "certChainFilePath");
```

Comment:
Please use more verbose error messages, as in the client file.

Reply:
resolved? (also, should I remove the checkNotNull(port...), since its an int-- just like in Client?

Reply:
Yes, please remove that checkNotNull and resolve this comment.